src/integration/resources/inprocess/coordinator.go

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) 2021  Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package inprocess contains code for spinning up M3 resources in-process for
+// the sake of integration testing.
+package inprocess
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/cenkalti/backoff/v3"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v2"
+
+	"github.com/m3db/m3/src/cmd/services/m3query/config"
+	"github.com/m3db/m3/src/integration/resources"
+	"github.com/m3db/m3/src/integration/resources/common"
+	nettest "github.com/m3db/m3/src/integration/resources/net"
+	"github.com/m3db/m3/src/query/generated/proto/admin"
+	"github.com/m3db/m3/src/query/generated/proto/prompb"
+	"github.com/m3db/m3/src/query/server"
+	xconfig "github.com/m3db/m3/src/x/config"
+)
+
+type coordinator struct {
+	cfg     config.Configuration
+	client  common.CoordinatorClient
+	logger  *zap.Logger
+	tmpDirs []string
```

Comment:
Instead of tracking this separately, is there any easy way to just glean this from the config? I.e. when you call updateCoordinatorFilepaths instead of returning a slice of tmpDirs, just modify the config directly and then at close time, iterate over the paths again and delete anything that looks like a tmp dir?

Reply:
That could work, but feels more cumbersome than just tracking the directories I created, tbh.